### PR TITLE
feat: adding all structs for easier bcs encoding-decoding.

### PR DESCRIPTION
**Pull request by Foivos**
https://axelarnetwork.atlassian.net/browse/AXE-4137

### CODE REVIEW

**Review by re1ro — APPROVED** [review]

**Review by npty — APPROVED** [review]

File: move/gas_service/Move.lock [hunk 1]
```diff
@@ -21,6 +21,6 @@ dependencies = [
 ]
 
 [move.toolchain-version]
-compiler-version = "1.25.1"
+compiler-version = "1.26.1"
```
Comment (milapsheth) [inline move/gas_service/Move.lock:24]: version should match `test.yaml` gh action. It should generally only be changed in a separate PR for clarity

File: move/squid/sources/squid/deepbook_v2.move [hunk 2]
```diff
@@ -25,7 +25,32 @@ module squid::deepbook_v2 {
...
+        pool_id: address,
+        has_base: bool,
+        min_output: u64,
+        base_type: String,
+        quote_type: String,
+        lot_size: u64,
+        should_sweep: bool,
+    }
+
+    fun new(data: vector<u8>): DeepbookV2SwapData {
```
Comment (milapsheth) [inline move/squid/sources/squid/deepbook_v2.move:39]:
```suggestion
    fun peel_swap_data(data: vector<u8>): DeepbookV2SwapData {
```
to mimic the pattern in the gateway

File: move/squid/sources/squid/transfers.move [hunk 3]
```diff
@@ -19,15 +19,51 @@ module squid::transfers {
     const EWrongSwapType: u64 = 0;
     const EWrongCoinType: u64 = 1;
 
+    public struct SuiTransferSwapData has drop {
+        swap_type: u8,
+        coin_type: String,
+        recipient: address,
+    }
+
+    public struct ItsTransferSwapData has drop {
```
Comment (milapsheth) [inline move/squid/sources/squid/transfers.move:28]:
```suggestion
    public struct ITSTransferSwapData has drop {
```

File: scripts/bcs.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    return {
+        DeepbookV2SwapData,
+        SuiTransferSwapData,
+        ItsTransferSwapData,
+    };
+}
+
+module.exports = {
+    axelarStructs: getAxelarStructs(),
+    squidStructs: getSquidStructs(),
```
Comment (milapsheth) [inline scripts/bcs.js:121]:
```suggestion
    gateway: getAxelarStructs(),
    squid: getSquidStructs(),
```
namespacing should be by contract name (or we can avoid extra namespacing completely)

File: scripts/publish-package.js [hunk 5]
```diff
@@ -71,6 +71,7 @@ if (require.main === module) {
         const privKey = Buffer.from(process.env.SUI_PRIVATE_KEY, 'hex');
         const keypair = Ed25519Keypair.fromSecretKey(privKey);
         const address = keypair.getPublicKey().toSuiAddress();
+        console.log(address);
```
Comment (milapsheth) [inline scripts/publish-package.js:74]: debug line?

File: scripts/tx-builder.js [hunk 6]
```diff
@@ -4,12 +4,27 @@ const {
     utils: { arrayify, hexlify },
 } = require('ethers');
 
+const objectCache = {};
+
+function updateCache(objectChanges) {
+    for (const change of objectChanges) {
+        if (!change.objectId) continue;
```
Comment (milapsheth) [inline scripts/tx-builder.js:11]: use braces with if blocks

**Review by milapsheth — APPROVED** [review]
